crates/transaction-pool/src/pool/mod.rs

```diff
@@ -500,6 +500,7 @@ where
 
         let mut listener = self.event_listener.write();
         discarded.iter().for_each(|tx| listener.discarded(tx));
+        drop(listener);
```

Comment:
pedantic style nit:
this should be in its own scope `{}` se we don't need to call drop